app/src/main/java/ingsw/group1/findmyphone/dialog/PasswordDialog.java

```diff
@@ -0,0 +1,108 @@
+package ingsw.group1.findmyphone.dialog;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.cryptography.PasswordManager;
+
+/**
+ * DialogFragment class displaying a Dialog used when setting the password.
+ *
+ * @author Riccardo De Zen.
+ */
+public class PasswordDialog extends DialogFragment {
+
+    private PasswordManager passwordManager;
+    private CharSequence previousPassword;
+    private View dialogRoot;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'dialogRoot' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5155842)